src/test/fixtures/mocks/intersection-observer.js

```diff
@@ -0,0 +1,122 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+/**
+ * Creating a mock intersection observer type because Flow's IntersectionObserver
+ * type is not completely correct in this version.
+ */
+type MockIntersectionObserver = {|
+  thresholds: number[],
+  root: HTMLElement | Document,
+  rootMargin: string,
+  observe: (HTMLElement) => void,
+  unobserve: (HTMLElement) => void,
+  disconnect: () => void,
+  takeRecords: () => void,
+|};
+
+/**
+ * Type of the item we are going to keep for tracking observers.
+ */
+type Item = {|
+  callback: (IntersectionObserverEntry[], MockIntersectionObserver) => void,
+  elements: Set<HTMLElement>,
+  created: number,
+|};
+
+/**
+ * Tracked observers during the testing.
+ */
+const observers: Map<MockIntersectionObserver, Item> = new Map();
+
+/**
+ * Call this function inside a `describe` block to automatically define the
+ * intersection observer.
+ */
+export function autoMockIntersectionObserver(autoTrigger?: boolean = true) {
+  beforeEach(() => {
+    (window: any).IntersectionObserver = jest.fn((cb, options = {}) => {
+      const item = {
+        callback: cb,
+        elements: new Set(),
+        created: Date.now(),
+      };
+
+      const instance: MockIntersectionObserver = {
+        thresholds: Array.isArray(options.threshold)
+          ? options.threshold
+          : [options.threshold ?? 0],
+        root: options.root ?? null,
+        rootMargin: options.rootMargin ?? '',
+        observe: jest.fn((element: HTMLElement) => {
+          item.elements.add(element);
+          if (autoTrigger) {
+            triggerSingleObserver(instance, item);
+          }
+        }),
+        unobserve: jest.fn((element: HTMLElement) => {
+          item.elements.delete(element);
+        }),
+        disconnect: jest.fn(() => {
+          observers.delete(instance);
+        }),
+        takeRecords: jest.fn(),
+      };
+
+      observers.set(instance, item);
+
+      return instance;
+    });
+  });
+
+  afterEach(() => {
+    delete (window: any).IntersectionObserver;
```

Comment:
Ah, good catch! I think I'm pretty sure that I wrote the line that clears the `observers` map, but it looks like somehow I managed to not commit it :sweat_smile: 

For checking if `observers.size === 0`, I tried this now and it seems like it's not clearing the observers during disconnect. But this could be a mock issue. I think it should be fine for now because we are not disconnecting any observers from timeline usually. So it should be safe for us. But I will investigate more and see after this PR.